### ISSUE TITLE
catkin_make failed with gtest 1.12.0-1

### ISSUE THREAD

**Issue by AngeloChen14** [ros-noetic-arch/ros-noetic-catkin#5]
After upgrade gtest to 1.12.0-1, the catkin build system is broken, `yay -S ros-noetic-catkin --rebuildtree` can't fix this problem.
build log: 
```
❯ catkin_make
Base path: /home/angelo/git/project/cyberc3_platooning
Source space: /home/angelo/git/project/cyberc3_platooning/src
Build space: /home/angelo/git/project/cyberc3_platooning/build
Devel space: /home/angelo/git/project/cyberc3_platooning/devel
Install space: /home/angelo/git/project/cyberc3_platooning/install
####
#### Running command: "cmake /home/angelo/git/project/cyberc3_platooning/src -DCATKIN_DEVEL_PREFIX=/home/angelo/git/project/cyberc3_platooning/devel -DCMAKE_INSTALL_PREFIX=/home/angelo/git/project/cyberc3_platooning/install -G Unix Makefiles" in "/home/angelo/git/project/cyberc3_platooning/build"
####
-- The C compiler identification is GNU 12.1.0
-- The CXX compiler identification is GNU 12.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using CATKIN_DEVEL_PREFIX: /home/angelo/git/project/cyberc3_platooning/devel
-- Using CMAKE_PREFIX_PATH: /home/angelo/git/project/cyberc3_platooning/devel;/opt/ros/noetic
-- This workspace overlays: /home/angelo/git/project/cyberc3_platooning/devel;/opt/ros/noetic
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.10.5", minimum required is "3")
-- Using PYTHON_EXECUTABLE: /usr/bin/python3
-- Using default Python package layout
-- Found PY_em: /usr/lib/python3.10/site-packages/em.py
-- Using empy: /usr/lib/python3.10/site-packages/em.py
-- Using CATKIN_ENABLE_TESTING: ON
-- Call enable_testing()
-- Using CATKIN_TEST_RESULTS_DIR: /home/angelo/git/project/cyberc3_platooning/build/test_results
-- Forcing gtest/gmock from source, though one was otherwise available.
-- Found gtest sources under '/usr/src/gmock': gtests will be built
-- Found gmock sources under '/usr/src/gmock': gmock will be built
CMake Warning at /usr/src/gmock/CMakeLists.txt:41 (project):
  VERSION keyword not followed by a value or was followed by a value that
  expanded to nothing.


CMake Warning at /usr/src/googletest/CMakeLists.txt:51 (project):
  VERSION keyword not followed by a value or was followed by a value that
  expanded to nothing.


-- Found Python: /usr/bin/python3.10 (found version "3.10.5") found components: Interpreter
-- Found Threads: TRUE
CMake Error at /usr/src/googletest/CMakeLists.txt:127 (set_target_properties):
  set_target_properties called with incorrect number of arguments.


CMake Error at /usr/src/googletest/CMakeLists.txt:129 (set_target_properties):
  set_target_properties called with incorrect number of arguments.


CMake Error at /usr/src/gmock/CMakeLists.txt:99 (set_target_properties):
  set_target_properties called with incorrect number of arguments.


CMake Error at /usr/src/gmock/CMakeLists.txt:102 (set_target_properties):
  set_target_properties called with incorrect number of arguments.


-- Using Python nosetests: /usr/bin/nosetests-3.10
-- catkin 0.8.10
-- BUILD_SHARED_LIBS is on
-- BUILD_SHARED_LIBS is on
```
The problem is gone after downgrade gtest to 1.11.0-4

**Comment by acxz** [comment]
Upstream issue: https://github.com/anthraxx/arch-pkgbuilds/issues/40

**Comment by acxz** [comment]
I have just created the [`gtest1.12`](https://aur.archlinux.org/packages/gtest1.12) package on the AUR and one can use that to install a version of gtest with the appropriate fix, while the arch package re incorporates the fix.

**Comment by AngeloChen14** [comment]
fixed with gtest 1.12.0-2
